L13-Python-Interceptors.md

```diff
@@ -0,0 +1,285 @@
+gRPC Python Interceptors
+----
+* Author(s): lidizheng
+* Approver: ericgribkoff
+* Status: In Review
+* Implemented in: Python
+* Last updated: January 17, 2019
+* Discussion at: https://groups.google.com/d/msg/grpc-io/7qIzyCPWbUA/1NKGXAQHCgAJ
+
+## Abstract
+
+This proposal is a hindsight design document that explains the design and implementation details of the interceptor functionality for gRPC Python, which merged in 2017. This gRFC serves as an effort to stabilize the interceptor APIs.
+
+
+## Background
+
+### The Need for Interceptors
+
+Interceptors are middleware which capable of manipulating or reacting to the transporting requests/responses throughout the life cycle of an RPC. It dramatically simplifies the implementation of tracing (See [grpc/grpc#8767](https://github.com/grpc/grpc/issues/8767)) and empowers gRPC developers to add general channel-level business logic.
+
+Interceptors have existed with gRPC for a while now, and its semantics were carried out across gRPC implementations like Java, Go and wrapper languages like Node.js, Ruby.
+
+### Stabilization Status
+
+| gRPC Language | Client Interceptor | Server Interceptor |
+| ------------- | ------------------ | ------------------ |
+| Python        | Experimental       | Experimental       |
+| Node.js       | Stable             | N/A                |
+| Java          | Stable             | Stable             |
+| Golang        | Experimental       | Experimental       |
+| C++           | N/A                | N/A                |
+| Ruby          | Experimental       | Experimental       |
+
+### Related Proposals
+
+* [Node.js Client Interceptor](https://github.com/grpc/proposal/blob/master/L5-node-client-interceptors.md)
+* [Ruby Client and Server Interceptors](https://github.com/grpc/proposal/pull/34)
+* [C# Client and Server Interceptors](https://github.com/grpc/proposal/pull/38)
+
+
+## Proposal
+
+### Interface: Client Call Details
+
+To intercept the RPC, gRPC needs to describe the upcoming RPC that enables the interceptor function to inspect and modify the call options. Hence we introduce the new interface `grpc.ClientCallDetails`, which contains five properties:
```

Comment:
This first sentence is a bit confusing to me. Perhaps "To intercept an RPC, we first require an interface by which a user may inspect and modify options pertaining to a specific call."?

Reply:
Updated, thank you for making it better.